javagen/src/main/java/com/azure/autorest/template/ModelTemplate.java

```diff
@@ -132,6 +132,15 @@ public final void write(ClientModel model, JavaFile javaFile) {
                 classBlock.privateStaticFinalVariable("Pattern KEY_ESCAPER = Pattern.compile(\"\\\\.\");");
             }
 
+            // If code is being generated with the behavior to return an empty byte array when the default value
+            // expression is null and the model has any array types that will need conversion within getter methods
+            // generate a static byte[] that will be returned instead of creating a new instance each get.
+            if (settings.isDefaultByteArrayReturnsEmptyArray()
+                && model.getProperties().stream().anyMatch(property -> property.getClientType() instanceof ArrayType
+                    && property.getWireType() != property.getClientType())) {
+                classBlock.privateStaticFinalVariable("byte[] EMPTY_BYTE_ARRAY = new byte[0]");
```

Comment:
Do we not have a constant in azure-core for this?

Reply:
Nothing that is public